conda_recipe_manager/types.py

```diff
@@ -35,9 +35,22 @@
 H = TypeVar("H", bound=Hashable)
 
 
-# All sentinel values used in this module should be constructed with this class, for typing purposes.
 class SentinelType:
-    pass
+    """
+    A single sentinel class to be used in this project, as an alternative to `None` when `None` cannot be used.
+    It is defined in a way such that SentinelType instances survive pickling and allocations in different memory
+    spaces.
+    """
+
+    def __eq__(self, o: object) -> bool:
+        """
+        Provides a pickle/thread-safe way to compare objects to the Sentinel Type. All sentinels are equal by simply
+        being sentinel objects.
+        :param o: The object to compare against.
+        :returns: True if the object is a SentinelType object.
+        """
+        # There is an implicit `isinstance(self, SentinelType) and ...` to this logic
+        return isinstance(o, SentinelType)
```

Comment:
```python
class X:
    def __new__(cls):
        global singleton
        try:
            return singleton
        except NameError:
            singleton = super().__new__(cls)
            return singleton

>>> x = X()
>>> x2 = X()
>>> x is x2
True
>>> import pickle
>>> x3 = pickle.loads(pickle.dumps(x2))
>>> x3 is x
True
```

Reply:
I think this is in-line with the libraries that @jezdez suggested. But it's also pretty simple, which I like. You've also reminded me that I forgot to add a unit test to pickle this type directly. Thanks!

Reply:
Let me take a stab at implementing this version.

Reply:
Had to make a few modifications to make the static analyzer and linter happy, but it appears to work!